### PR TITLE
[feat]: [CDS-50873]: Delete trigger event history when pipeline is deleted

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
## Describe your changes
Delete trigger event history when pipeline is deleted

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43304)

### CODE REVIEW

**Review by vcalasansh — DISMISSED** [review]
> Reviewed 4 of 5 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43304)* status: 6 file(s) to review. 1 of 2 approvals obtained.

File: pipeline-service/service/src/test/java/io/harness/pms/event/entitycrud/PipelineEntityCRUDStreamListenerTest.java [hunk 1]
```diff
@@ -62,6 +63,7 @@
 @OwnedBy(PIPELINE)
 public class PipelineEntityCRUDStreamListenerTest extends CategoryTest {
   @Mock private NGTriggerService ngTriggerService;
+  @Mock private NGTriggerEventsService ngTriggerEventsService;
```
Comment (sribalij) [inline pipeline-service/service/src/test/java/io/harness/pms/event/entitycrud/PipelineEntityCRUDStreamListenerTest.java:66]: Can you add a few Junits to test it?

**Review by sribalij — CHANGES_REQUESTED** [review]

**Review by rathodmeetsatish — COMMENTED** [review]
> Reviewed 4 of 5 files at r1, 1 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43304)* status: 7 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/test/java/io/harness/pms/event/entitycrud/PipelineEntityCRUDStreamListenerTest.java` line 66 at r2](https://reviewable.io/reviews/harness/harness-core/43304#-NNBXEasALV8j_HCvwpR-r2-66:-NNGi8FWBp9VBswTeyOG:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d83e4a60b6183c296a7bd83893f0253011e4ee4e/pipeline-service/service/src/test/java/io/harness/pms/event/entitycrud/PipelineEntityCRUDStreamListenerTest.java#L66)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

Can you add a few Junits to test it?
</blockquote></details>

Done.

**Review by sribalij — APPROVED** [review]
> Reviewed 4 of 5 files at r1, 1 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43304)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by rathodmeetsatish** [comment]
trigger messagecheck

**Review by vcalasansh — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43304)* status: 1. unresolved discussions.

**Review by vcalasansh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43304)* status: 1. unresolved discussions.

**Review by sribalij — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43304)* status: :shipit: complete! ✓

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 4 of 5 files at r1, 1 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43304)* status: :shipit: complete! ✓

**Pull request merged**